Assignment2/Logger/BaseLoggerMixins.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace Logger
+{
+    public static class BaseLoggerMixins
+    {
+        public static void Error(this BaseLogger baseLogger, string message, params object[] messageArgs)
```

Comment:
While the type of the first parameter is `BaseLogger`, the parameter will never actually be just a `BaseLogger` (because it is abstract and you can't instantiate an abstract class) but instead it will be a concrete class that inherits from `BaseLogger` like `FileLogger`. So `baseLogger` as a variable name is misleading.
```suggestion
        public static void Error(this BaseLogger logger, string message, params object[] messageArgs)
```